pkg/authentication/handlers.go

```diff
@@ -123,6 +169,37 @@ func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
 	}
 
 	_ = json.NewEncoder(w).Encode(tokens)
+
+}
+
+func (a *API) nonceCheck(r *http.Request, idToken *Principal) error {
```

Comment:
nitpick: I prefer `checkNonce`, that way is it is clearer that this function checks something